processor/src/coin/bitcoin.rs

```diff
@@ -0,0 +1,380 @@
+use async_trait::async_trait;
+use bitcoin_hashes::hex::{FromHex, ToHex};
+use std::str::FromStr;
+use transcript::RecommendedTranscript;
+
+use frost::{
+  curve::Secp256k1,
+  ThresholdKeys,
+};
+
+use bitcoin::{
+  Block as BBlock, OutPoint,
+  util::address::Address, consensus::encode,
+  Txid, schnorr::TweakedPublicKey,
+  XOnlyPublicKey, SchnorrSighashType,
+  psbt::{PartiallySignedTransaction,PsbtSighashType},
+};
+
+use bitcoin_serai::{
+  rpc::Rpc,
+  crypto::make_even,
+  wallet::SpendableOutput,
+  transactions::{TransactionMachine, SignableTransaction as BSignableTransaction},
+};
+
+use k256::{
+  ProjectivePoint, Scalar,
+  elliptic_curve::sec1::{ToEncodedPoint, Tag},
+};
+use crate::{
+  coin::{CoinError, Block as BlockTrait, OutputType, Output as OutputTrait, Coin}
+};
+
+#[derive(Clone, Debug)]
+pub struct Block([u8; 32], BBlock);
```

Comment:
You don't need the `[u8; 32]` since you can get the hash from the block itself.

Reply:
Actually, you shouldn't even need the tuple. You should be able to implement BlockTrait directly on Block.